e2e/main.py

```diff
@@ -69,26 +51,50 @@ def test_e2e(self):
             checker_dir=os.path.join(PROJECT_ROOT, "generated/michelson"),
             oracle=oracle.context.address,
             ctez=ctez["fa12_ctez"].context.address,
+            num_blocks_wait=100,
+            ttl=MAX_OPERATIONS_TTL,
         )
 
         print("Deployment finished.")
-
-        (
-            checker.create_burrow((1, None))
-            .with_amount(10_000_000)
-            .as_transaction()
-            .autofill(branch_offset=1)
-            .sign()
-            .inject(min_confirmations=1, time_between_blocks=3)
-        )
-
-        (
-            checker.mint_kit((1, 1_000_000))
-            .as_transaction()
-            .autofill(branch_offset=1)
-            .sign()
-            .inject(min_confirmations=1, time_between_blocks=3)
-        )
+        gas_costs = {}
+
+        def call_endpoint(contract, name, param, amount=0):
+            ret = (
+                getattr(contract, name)(param)
+                .with_amount(amount)
+                .as_transaction()
+                .autofill(ttl=MAX_OPERATIONS_TTL)
+                .sign()
+                .inject(
+                   min_confirmations=1,
+                   num_blocks_wait=100,
+                   max_iterations=WAIT_BLOCK_ATTEMPTS,
+                   delay_sec=WAIT_BLOCK_DELAY
+                )
+            )
+            return ret
+
+        def call_checker_endpoint(name, param, amount=0):
+            ret = call_endpoint(checker, name, param, amount)
+            gas_costs[name] = ret["contents"][0]["gas_limit"]
```

Comment:
FYI - Since we are actually calling the endpoints we can use the OperationResult to get the `consumed_gas` (pretty sure that is the name of the field). I've noticed that it is sometimes quite a bit less than the estimated `gas_limit` when working with a fresh contract. 

The approach here is a more conservative estimate though, so I'm also happy leaving it as-is.

Reply:
I would actually prefer using the real `consumed_gas` field, however, `ret` there does not contain a field like that.

I think it should be possible to find the operation group and extract the information from there, but I'm afraid that'd make this code a bit more complex. So I think we can merge this one, but I'd be happy to change it to `consumed_gas` if we find an easy way to get it.